programFiles/may.cpp

```diff
@@ -0,0 +1,327 @@
+#include<iostream>
+#include<vector>
+#include<fstream>
+#include<string>
+#include<sstream>
+#include<cmath>
+double cRound(double number){
+	if(fmod(number,10)>=5)return (((int)number/1)+1);
+	else return ((int)number/1);
+}
+//N-число строк M-число столбцов
+class matrix{
+		int Nsize;
+		int Msize;
+		std::vector<std::vector<double>> matrixData;
+	public:
+		//Конструкторы. 1) Создание пустой (нулевой) матрицы NxM 2) Считывание матрицы из файла
+		matrix(int N,int M){
+			Nsize = N;
+			Msize = M;
+			std::vector<std::vector<double>> temp(N, std::vector<double> (M,0));
+			matrixData = temp;
+		}
+		matrix(std::string filename){
+			std::ifstream file(filename);
+			if(file.is_open()){
+				std::string input;
+				int array[2] = {0,0};
+				for(int i=0;i<2;i++){
+					if(i==0)std::getline(file,input,' ');
+					if(i==1)std::getline(file,input,'\n');
+					array[i] = stod(input);
+				}
+				input.clear();
+				
+				Nsize = array[0];
+				Msize = array[1];
+				
+				std::string check;
+				while(!file.eof()){std::getline(file,check);input=input + check;check.clear();}
+
+				std::stringstream str(input);
+				
+				std::vector<std::vector<double>> temp(array[0], std::vector<double> (array[1],0));
+
+				
+				matrixData = temp;
+				
+				for(int i=0;i<matrixData.size();i++){
+					std::string tempNumber;
+					for(int j=0;j<matrixData[i].size();j++){
+						
+						str >> tempNumber;
+						matrixData[i][j] = stod(tempNumber);
+						tempNumber.clear();
+
+					}
+
+				}
+			}
+			file.close();
+		}
+		//
+		
+		//Создание нулевой, либо единичной матрицы NxM
+		void static makeMatrix(matrix obj,int N,int M,char F){
+			obj.Nsize = N;
+			obj.Msize = M;
+			if (F=='E'){
+				std::vector<std::vector<double>> temp(N, std::vector<double> (M,0));
+				for(int i = 0;i<temp.size();i++){
+					for(int j = 0;j<temp[i].size();j++){
+						if(i==j) 	temp[i][j] = 1;
+					}
+				}
+				obj.matrixData = temp;
+			}
+			if (F=='O'){
+				std::vector<std::vector<double>> temp(N, std::vector<double> (M,0));
+				obj.matrixData = temp;
+			}
+		}
+		//
+		
+		//Запись матрицы в файл
+		void matrixToFile(std::string filename){
+			std::ofstream file(filename);
+			if(file.is_open()){
+				file << Nsize  <<' '<<Msize<<'\n';
+				for(int i=0;i<Nsize;i++){
+					for(int j=0;j<Msize;j++){
+						file << matrixData[i][j] <<' ';
+					}
+				}
+			}
+			file.close();
+		} 
+		//
+		
+		//Перегрузка операторов +,-,*(на число и на матрицу)
+		matrix operator+(matrix r){

```

Comment:
In the operator+ and operator- operators, you can get rid of repetitive code, for example, allocate a loop into a separate function.